shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -127,6 +127,16 @@ public boolean setSelection(int start, int end) {
         return result;
     }
 
+    private void deleteLastCharacter(int selStart, int selEnd) {
+        String str = mEditable.toString();
+        //calc emoji chatacter length (most is 2,and it's can't deal with charactors whick too long)
```

Comment:
This will work for simple emojis, but will not work for something like 👨‍👩‍👧‍👧, which is made up of 4 16bit emoji codepoints as well as 3 8bit zero-width-joiner codepoints.

I like the idea of handling this in the platform code though, as it is currently not (yet) possible to obtain the lengths of extended grapheme clusters in a dart string.